test.py

```diff
@@ -911,6 +962,9 @@ def _prepare_pytest_params(self, options: argparse.Namespace):
             "junit_family=xunit2",
             "--junit-xml={}".format(self.xmlout),
             "-rs"]
+        if self.case:
+            self.args.extend(["-k", self.case])
```

Comment:
Why "-k" and not just give the file name and case name?
I'm worried that it can pick up similarly-named tests from other files as well, and other strange things (I don't know exactly what "-k" does...).